src/bundle-manifest.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * A bundle is a grouping of files which serve the need of one or more
+ * entrypoint files.
+ */
+export interface IBundle {
+  entrypoints: Set<URL>;
+  files: Set<URL>;
+}
+
+/**
+ * A bundle strategy function is used to transform an array of bundles.
+ */
+export type IBundleStrategy = (bundles: IBundle[]) => IBundle[];
+
+/**
+ * A mapping of URLs to their full set of transitive dependencies, such that
+ * `a->b, b->c, b->d, e->f` would be represented `{a:[b,c,d], e:[f]}`
+ */
+export type IFileTransDepsIndex = Map<URL, Set<URL>>;
+
+/**
+ * Defining this URL type to make it clear which strings represent URLs.
+ */
+export type URL = string;
+
+/**
+ * Standard implementation of IBundle.
+ */
+export class Bundle implements IBundle {
```

Comment:
Learning that you can give compatible alternate implementations to stand in for typed values changed my perception of Typescript.  I assumed Interfaces were the only means to do that.  I like what I have learned.  Ditching IBundle.